### ISSUE TITLE
[a11][VoiceOver/Talkback] Dismissable ModalBarrier is not announced as a button

### ISSUE THREAD

**Issue by Jerem42** [flutter/flutter#158164]
### Steps to reproduce

1. Have an app with a popupMenu
2. Enable VoiceOver / Talkback
3. Swipe until you get Dismiss Menu semantic

### Expected results

Element should be labelled as button by VoiceOver/Talkback as it has an onTap action.

### Actual results

Element is not labelled as button

### Code sample

<details open><summary>Code sample</summary>

```dart
void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      supportedLocales: <Locale>[
        Locale('en', 'US'), // English
      ],
      home: Scaffold(
        body: Center(
          child: TextButton(
              child: Text('Hello, World!'),
              onPressed: () {
                showMenu<String>(
                  shape: const RoundedRectangleBorder(
                      borderRadius: BorderRadius.all(Radius.circular(16.0))),
                  context: context,
                  items: [
                    PopupMenuItem(child: Text('Menu1')),
                    PopupMenuItem(child: Text('Menu2'))
                  ],
                  position: RelativeRect.fromLTRB(
                    0,
                    0,
                    MediaQuery.of(context).size.width / 2,
                    MediaQuery.of(context).size.height / 2,
                  ),
                );
              }),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/7c9983ab-13eb-4886-9733-9209744b3474

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.19.3, on macOS 14.6.1 23G93 darwin-arm64, locale fr-FR)
    • Flutter version 3.19.3 on channel stable at /Users/jeremy_bastien/fvm/versions/3.19.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ba39319843 (8 months ago), 2024-03-07 15:22:21 -0600
    • Engine revision 2e4ba9c6fb
    • Dart version 3.3.1
    • DevTools version 2.31.1

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/jeremy_bastien/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/jeremy_bastien/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode 15/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] IntelliJ IDEA Ultimate Edition (version 2024.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.5)
    • IntelliJ at /Users/jeremy_bastien/Applications/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 81.0.2
    • Dart plugin version 241.18968.26

[✓] Connected device (2 available)
    • macOS (desktop)                  • macos                     • darwin-arm64   • macOS 14.6.1 23G93 darwin-arm64
    • Chrome (web)                     • chrome                    • web-javascript • Google Chrome 130.0.6723.92

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by darshankawar** [comment]
@Jerem42 
Using your code and running on latest stable, tapping on hello world text isn't triggering the popup menu for some reason. Can you try the same or make updates to the code that will show it ?

**Comment by Jerem42** [comment]
I just edited the code sample to make it work.

**Comment by darshankawar** [comment]
Thanks for the update. Upon running on Android and iOS, it reads `dismiss menu, double tap to activate`. Although upon focusing on menu inside dismissable, it reads correctly as button.

```
stable : 3.24.3
master : 3.27.0-1.0.pre.367
```

**Comment by chunhtai** [comment]
having a ontap doesn't necessary mean it is a button. We have to double check with native behavior

**Comment by eMxPi** [comment]
hello any update on this issue ?
